### PR TITLE
Fix issue #2374 Comments of added Nodes are ignored in LexicalPreserv…

### LINKED ISSUE #2374: Comments of added Nodes are ignored in LexicalPreservingPrinter
Given this example code
```java
public class Bar {
    public void foo() {
        System.out.print("Hello");
    }
}
```
If I parse it with javaparser and setup the AST with the LPP and adding a new node as
```java
Statement stmt = StaticJavaParser.parseStatement("System.out.println(\"World!\");");
stmt.setLineComment("Example comment");
```
the result of `LexicalPreservingPrinter#print` will look like this:
```java
public class Bar {
    public void foo() {
System.out.print("Hello!");
System.out.println("World!");
    }
}
```
I have taken a look into the code and found out, that the CSM does not fully support comments as `CsmComment`. Is there any plan to support these comments or is it even a bug? I found a work around for it by adding the new node to the AST first and afterwards adding the comment to the added node. Thank you!

### PR DESCRIPTION
**Pull request by jlerbsc**
…ingPrinter

Fixes #2374 .

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalDifferenceCalculator.java [hunk 1]
```diff
@@ -146,6 +149,19 @@ private void calculatedSyntaxModelForNode(CsmElement csm, Node node, List<CsmEle
...
+                // fix issue #2374
+                // Add node comment if needed (it's not very elegant but it works)
+                // We need to be sure that the node is an ExpressionStmt because we can meet
+                // this class definition
+                // a line comment <This is my class, with my comment> followed by
+                // class A {}
+                // In this case keyworld [class] is considered as a token and [A] is a child element
+                // So if we don't care that the node is an ExpressionStmt we could try to generate a wrong definition
+                // like this [class // This is my class, with my comment A {}]
+                if (node.getComment().isPresent() && node instanceof ExpressionStmt) {
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalDifferenceCalculator.java:161]: What is the defining characteristic of nodes like ExpressionStmt that make them need behaviour like this? Maybe we can make that a property of the node, so we can avoid node specific exceptional logic here.

Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalDifferenceCalculator.java:161]: @jlerbsc ?

Comment (jlerbsc) [reply javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalDifferenceCalculator.java:161]: I think it's more specific to lexical printer than expression statement.

Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalDifferenceCalculator.java:161]: Okay, but do you think there are other nodes likes this? If not, it's okay.

Comment (jlerbsc) [reply javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalDifferenceCalculator.java:161]: Sincerely, I don't know! I don't know enough JavaParser to answer this question. Maybe the author could answer it.

Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalDifferenceCalculator.java:161]: Let's just merge it then :-)

**Pull request merged**